src/sgame/sg_bot_nav.cpp

```diff
@@ -412,9 +412,38 @@ static bool BotShouldJump( gentity_t *self, const gentity_t *blocker, const glm:
 	return blocker->s.modelindex == BA_A_BARRICADE || trace.fraction == 1.0f;
 }
 
-// try to find a path around the obstacle by projecting 5
+// Determine a new direction, which may be unchanged (forward) or sideway. It
+// does try the following every second: one time left, one time right and one
+// time forward until they find their way.
+static void BotFindDefaultSteerTarget( gentity_t *self, glm::vec3 &dir )
+{
+	// turn this into a 2D problem
+	dir[2] = 0;
+	// a vector 90° from dir, pointing right and of same magnitude
+	glm::vec3 right = glm::vec3(dir[1], -dir[0], 0);
+
+	int clientNum = self - g_entities;
+	// We add a delta determined from clientNum to avoid bots
+	// dancing in sync
+	int time = (110 * clientNum + self->client->time1000) % 1000;
+
+	// 280 instead of 700/2=350 because we want some left/right
+	// asymetry in case it helps
+	bool invert = time < 280;
+	// half of the time we go straight instead
+	bool sideway = time > 700;
+
+	// 2*right + 1*dir means an angle of atan(2/1) = 63°. This
+	// seems about right
+	dir += right * (invert ? -1.0f : 1.0f)
```

Comment:
I think `if (sideways) { ... }` would be more readable. Also it should be "sideways" not "sideway".

LGTM

Reply:
updated